pkg/crinit/aggregated/aggregated.go

```diff
@@ -0,0 +1,399 @@
+	if err != nil {
+		return err
+	}
+
+	credentials, err := opts.GenerateCredentials(cmdOut, svc.Name, ips, hostnames,
+		false, false)
+	if err != nil {
+		return err
+	}
+
+	err = opts.CreateAPIServerCredentialsSecret(hostClientset, credentials)
+	if err != nil {
+		return err
+	}
+
+	var pvc *v1.PersistentVolumeClaim
+	pvc, err = opts.CreatePVC(cmdOut, hostClientset, svc.Name)
+	if err != nil {
+		return err
+	}
+
+	err = opts.CreateAPIServer(cmdOut, hostClientset, false, false, ips, pvc)
+	if err != nil {
+		return err
+	}
+
+	_, err = createAPIService(cmdOut, apiSvcClientset, opts,
+		util.GetCAKeyPair(credentials).Cert)
+	if err != nil {
+		return err
+	}
+
+	err = createRBACObjects(cmdOut, hostClientset, opts)
+	if err != nil {
+		return err
+	}
+
+	err = opts.UpdateKubeconfig(cmdOut, pathOptions, svc, ips, hostnames,
+		credentials)
+	if err != nil {
+		return err
+	}
+
+	return opts.WaitForAPIServer(cmdOut, hostClientset, pathOptions, ips,
+		hostnames, svc)
+}
+
+// createRbacObjects handles the creation of all the RBAC objects necessary
+// to deploy the cluster registry in aggregated mode
+func createRBACObjects(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) error {
+
+	fmt.Fprintf(cmdOut, "Creating RBAC objects...")
+
+	// Create a Kubernetes service account in your namespace
+	glog.V(4).Infof(
+		"Creating service account %v for cluster registry apiserver in the host cluster",
+		serviceAccount)
+
+	sa, err := createServiceAccount(cmdOut, clientset, opts)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to create service account %v: %v", sa, err)
+		return err
+	}
+
+	glog.V(4).Info("Successfully created service account")
+
+	// Create a Kubernetes cluster role for the operations you want to allow
+	// on your resources
+	glog.V(4).Infof("Creating cluster role %v", clusterRole)
+
+	cr, err := createClusterRole(cmdOut, clientset, opts)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to create cluster role %v: %v", cr, err)
+		return err
+	}
+
+	glog.V(4).Info("Successfully created cluster role")
+
+	// Create a Kubernetes cluster role binding from the default service account
+	// in your namespace to the cluster role you just created
+	glog.V(4).Infof("Creating cluster role binding %v", clusterRole)
+
+	crb1, crb2, err, err2 := createClusterRoleBindings(cmdOut, clientset, opts)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to create cluster role binding %v: %v", crb1, err)
+		return err
+	} else if err2 != nil {
+		glog.V(4).Infof("Failed to create cluster role binding %v: %v", crb2, err2)
+		return err
+	}
+
+	fmt.Fprintln(cmdOut, " done")
+	glog.V(4).Info("Successfully created cluster role binding")
+	return nil
+}
+
+// createServiceAccount handles the creation of the service account for the
+// the cluster registry to be used with RBAC.
+func createServiceAccount(cmdOut io.Writer, clientset client.Interface,
+	opts *aggregatedClusterRegistryOptions) (*v1.ServiceAccount, error) {
```

Comment:
I think it would be better to just pass `name` and `dryRun`: it makes the coupling here looser. Consider that for other method calls too, though I can understand if you decide that the parameter lists are too long